### PR TITLE
AFNetworking 4

### PR DESCRIPTION
**Pull request by jshier**
This PR tracks the changes for AFNetworking 4.

Issues remaining:
- [x] Failing encoding test on Catalyst.
- [x] Failing trust tests on Catalyst.

These failures are strange, all macOS and iOS tests pass.

### CODE REVIEW

**Comment by Kaspik** [comment]
Please also update the podspec to fix warnings for projects - https://github.com/AFNetworking/AFNetworking/pull/4522/files#diff-1f8ef09d890801d130cd1ee773e1cb2eR38

**Comment by jshier** [comment]
@Kaspik How would I make those includes conditional for all deployment targets?

**Comment by Kaspik** [comment]
They are iOS / watchOS only. They were renamed, but it's still the same SDK, so projects will handle that correctly afaik.

**Comment by jshier** [comment]
Right, but the rename is only valid when built using the 13.4 SDK right?

**Comment by Kaspik** [comment]
Tests here should catch that because they are not using XCode 11.4 (or if you have one), or we could create dummy project for that - but I hope it's backwards compatible, I don't want to run another branch for years to silence the warning. 😭

**Comment by jshier** [comment]
There is no warning in the project locally, likely due to the fact that our deployment targets are lower. If the rename works with Xcode 11+ and doesn't require 11.4, then I'm fine doing the rename. I'll try it out locally and see.

**Comment by jshier** [comment]
Building using Xcode 11.3 fails if we only import `CoreServices.h`.

**Comment by Kaspik** [comment]
Allright, let's release 4.0.0 and then we can separate branch/PR/release after that with this change. Thanks for testing!

**Comment by jshier** [comment]
I do wonder if there's some condition we can check to avoid importing MobileCoreService on newer versions. I'll look into it.

**Comment by bofeizhu** [comment]
Hi @jshier, it looks like Xcode 11.4 is not yet available on Travis.

**Comment by jshier** [comment]
@bofeizhu Thanks. I'm not sure that's my only problem but it certainly doesn't help.

**Comment by Kaspik** [comment]
Seeing all the Travis problems, did you consider using Github Actions?

**Comment by jshier** [comment]
@Kaspik Travis wasn't really the problem, more my confusion around their environment and the old Fastlane stuff that this library uses. However, that does remind me that Travis doesn't yet support 10.15 while GitHub does, so switching would allow us to test Catalyst. And I don't think it's a problem that they only support Xcode 11, as Apple will start requiring that in June (was April) anyway.

**Comment by Kaspik** [comment]
Cool! Checkout MessageKit setup I did for Github Actions if you want - https://github.com/MessageKit/MessageKit/pull/1297

**Comment by jshier** [comment]
There we go. And much faster too.

**Comment by Kaspik** [comment]
Yeah Github Actions are pretty awesome, yet expensive for macOS private repos.

**Comment by jshier** [comment]
@kinarobin @ElfSundae @Kaspik Please take another look on master for any last minute fixes or changes you'd like to see. I'm preparing the 4.0 release now.

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+//
+
+import PackageDescription
+
+let package = Package(name: "AFNetworking",
+                      platforms: [.macOS(.v10_10),
+                                  .iOS(.v9),
+                                  .tvOS(.v9),
+                                  .watchOS(.v2)],
+                      products: [.library(name: "AFNetworking",
```
Comment (Kaspik) [inline Package.swift]: Was there any reason why AFNetworking is not enabled to be dynamic library?

`.library(name: "AFNetworking",, type: .dynamic, targets: ["AFNetworking"])`

cc @joshfriend

Comment (jshier) [reply Package.swift]: That setting should be controlled by consumers, not the library.

Comment (Kaspik) [reply Package.swift]: Yeah, if consumer uses Package.swift and adds AFNetworking as a dependency, then it's possible.

But projects using SPM through XCode don't have this option yet and it gets imported always as a static library... 😞 
https://forums.swift.org/t/how-to-link-a-swift-package-as-dynamic/32062

Screenshot 2020-03-30 at 00 43 28


Thanks tho.

Comment (jshier) [reply Package.swift]: Yeah, unfortunately Xcode integrations are always static, no matter what the library says.

**Pull request merged**